utils/run_sample_ci.py

```diff
@@ -34,6 +36,11 @@ def get_secrets_and_launch(parsed_commands):
     sample_private_key = ""
     sample_custom_authorizer_name = ""
     sample_custom_authorizer_password = ""
+    sample_certificate_x509 = ""
+    sample_private_key_x509 = ""
+    sample_ca_x509 = ""
+    sample_endpoint_x509 = ""
+    sample_role_alias_x509 = ""
```

Comment:
The run_sample_ci.py script becomes complicated and not extendable at this point. If we had new sample with new arguments in the future, we had to go back here and update.
The solution is a working one, while if we had time frame for this, probably using a config file or passing in parameter lists instead of hardcoded the parameters in the script would make it more compatible for future samples. 

Not for this PR, but we can also update the aws-crt-builder to include the run_sample_ci, so that builder would take care of everything, and we would not need to worry about installation on GitHub CI as well. (Sorry this would be a totally new feature request and I don't think we have time frame for it now, as we would need test the builder.pyz against all c-* and crt repos if we update the builder. I just want to mention it could be a way to go in the future.)

Reply:
Yeah, I totally agree. The `run_sample_ci` script initially was small and lean, but as I needed to run more samples, it has ballooned up in both size and complexity. The problem is we have some samples that require very specific arguments and data passed which are not really easily made generic, so I just added them as I went. I do agree though that it is not scalable and anytime we need to add a new sample that has different/complex arguments, the entire script has to be modified across all 4 SDKs.

Using a config file or passing in the parameters is a good idea. I think then it could just read that file, pull what it needs (either from secrets or otherwise - the config/parameter could probably specify) and then just run the sample. Ideally this config file could also handle a lot of the complexity around how to pass parameters on some of the more complex samples (PKCS11, X509, etc) as well, allowing for the script to be lean.

For the builder, that would be a good plan. There is a few scripts we could move to the builder to simplify things and make it where there is just a single place we have to made edits rather than across 4 SDKs. As you mentioned though, this would be outside of the time frame for now but would be a good addition/change for the future.

________

Regardless, I agree some refactoring work here on the `run_sample_ci.py` script would be ideal to reduce complexity and make it extendable.